ObjectCreation/AeroFoil.FCMacro

```diff
@@ -0,0 +1,1747 @@
+
+  Contact Information :-
+  Email :  carlo.melwyn@outlook.com
+  FreeCAD UserTalk :  http://www.freecadweb.org/wiki/index.php?title=User:Melwyncarlo
+
+##########################################################################################
+#####				L I C E N S E					     #####
+##########################################################################################
+
+
+
+  To use this macro, the steps to be followed are simple and straightforward :
+  follow the instructions in the respective dialog boxes, fill in the relevant inputs, 
+  and navigate accordingly. In case of error or warning, you will automatically be 
+  notified the same. In case you are notified to report an unexpected error, 
+  communicate the error by mentioning the FreeCAD version, tracing the steps 
+  taken, and mentioning whether (and how much) or not any ouput was generated.
+
+  Note (1)	Performing the macro operation with custom points and refinement 
+		produces no visible changes.
+  Note (2)	The AeroFoil object properties are only visible on the FreeCAD 
+		software version 0.19. On older versions, you will be shown 
+		a warning on the console.
+
+
+
+  The AeroFoil macro was developed and tested on a platform containing the 
+  following system and FreeCAD software specifications : 
+
+  - OS			: Ubuntu 18.04.5 LTS (LXDE/Lubuntu)
+  - Word size of OS	: 64-bit
+  - Word size of FreeCAD: 64-bit
+  - Version		: 0.19
+  - Build type		: Release
+  - Branch		: unknown
+  - Hash		: 32200b604d421c4dad527fe587a7d047cf953b4f
+  - Python version	: 3.6.9
+  - Qt version		: 5.9.5
+  - Coin version	: 4.0.0a
+  - OCC version		: 7.3.0
+  - Locale		: English/UnitedKingdom (en_GB)
+
+"""
+
+
+
+__Title__ = "AeroFoil"
+__Author__ = "Melwyncarlo"
+__Version__ = "1.0.0"
+__Date__ = "2021-02-26"
+__Comment__ = "AeroFoil creates airfoil curves and faces using pre-defined models, algebraic functions, and DAT or CSV Files"
+__Web__ = "https://github.com/melwyncarlo/AeroFoil"
+__Wiki__ = "http://www.freecadweb.org/wiki/index.php?title=Macro_AeroFoil"
+__Icon__ = "AeroFoil_UI_Files/AeroFoil.png"
+__Help__ = "Click on the AeroFoil button/macro, and follow the instructions in the subsequent dialog boxes."
+__Status__ = "stable"
+__Requires__ = "Freecad >= v0.17"
+__Communication__ = "https://github.com/melwyncarlo/AeroFoil/issues"
+__Files__ = "AeroFoil_UI_Files/AeroFoil_Dialog_1.ui, AeroFoil_UI_Files/AeroFoil_Dialog_2a.ui, AeroFoil_UI_Files/AeroFoil_Dialog_2b.ui, \
+AeroFoil_UI_Files/AeroFoil_Dialog_2c.ui, AeroFoil_UI_Files/AeroFoil_Dialog_2d.ui, AeroFoil_UI_Files/AeroFoil_Dialog_2d1a.ui, \
+AeroFoil_UI_Files/AeroFoil_Dialog_2d1b.ui, AeroFoil_UI_Files/AeroFoil_Dialog_2cd2.ui, AeroFoil_UI_Files/AeroFoil_Dialog_3.ui, \
+AeroFoil_UI_Files/AeroFoil_Math_Functions_Box.ui, AeroFoil_UI_Files/AeroFoil_mfb_img.gif"
+
+
+
+# Library Imports
+# ------------------------------------------------------------------------------------------------
+
+import FreeCAD as App
```

Comment:
Yes, I'll do that too.

Reply:
Should I use "app" and "gui" or "App" and "Gui"? 
I'm confused because Python is case-sensitive, 'App' and 'Gui' are pre-defined variables, and also on forums, they tend to use the latter.

Reply:
Never mind, I'll use the lower case variants.